docs/api.md

```diff
@@ -4,211 +4,411 @@ title: single-spa API
 sidebar_label: API
 ---
 
-The single-spa library does not `export default`, but instead exports named functions and variables.
-What this means is you can use the api in two ways:
+single-spa exports named functions and variables rather than a single default export.
+This means importing must happen in one of two ways:
 
 ```js
+import { registerApplication, start } from 'single-spa';
+// or
 import * as singleSpa from 'single-spa';
-// OR
-import {registerApplication, start} from 'single-spa';
 ```
 
 ## registerApplication
-`registerApplication(name, whereToGetApplication, activeWhen)` is the most important api your single spa config will use.
-It is described in detail inside of the [single-spa-config.md docs](single-spa-config.md#registering-applications)
+
+<!-- ```ts
+registerApplication: void (
+	appName: string,
+	applicationOrLoadingFn: () => Promise | Function,
+	activityFn: (location) => boolean,
+	customProps?: Object = {}
+) 
+``` -->
+
+`registerApplication` is the most important api your single spa config will use. Use this function to register any application within single-spa.
+
+Note that if an application is registered from within another application, that no hierarchy will be maintained between the applications.
+
+> It is described in detail inside of the [single-spa-config.md docs](single-spa-config.md#registering-applications)
+
+<h3>arguments</h3>
+
+<dl class="args-list">
+	<dt>appName: string</dt>
+	<dd>App name that single-spa will register and reference this application with, and will be labelled with in dev tools.</dd>
+	<dt>applicationOrLoadingFn: () => &lt;Function | Promise&gt;</dt>
+	<dd>Must be a loading function that either returns the resolved application or a promise.</dd>
+	<dt>activityFn: (location) => boolean</dt>
+	<dd>Must be a pure function. The function is called with <code>window.location</code> as the first argument <!-- TODO: any only? --> and should return a truthy value whenever the application should be active.</dd>
+	<dt>customProps?: Object = {}</dt>
+	<dd>Will be passed to the application during each lifecycle method.</dd>
+</dl>
+
+<h3>returns</h3>
+
+`undefined`
 
 ## declareChildApplication
-`declareChildApplication` is deprecated use `registerApplication` instead
+
+<p class="deprecated">Deprecated</p>
+
+Use [`registerApplication`](#registerapplication) instead.
 
 ## start
```

Comment:
this all looks amazing! however, I noticed that the right sidebar isn't picking up on the headers anymore on your example repo. Wonder what changed?

Reply:
Oh weird, I hadn't caught that. I'll look into why.

Reply:
Turns out it was this HTML comment here that was borking things. https://github.com/CanopyTax/single-spa.js.org/pull/48/files#diff-0f426376b964e609c5c4618eb8bed3b2R18

I'll remove it since it wasn't adding anything, just was an attempt at another style of displaying the args/returns.